File: seeker/snippet/bubble_sort.go
//date: 2021-12-13T17:06:04Z
//url: https://api.github.com/gists/81b68653ece8601c27fcdeee01c5c392
//owner: https://api.github.com/users/iulspop

package bubble_sort

func BubbleSort(numbers []int) []int  {
	unsortedUntilIndex := len(numbers) - 1
	swap := true

	for swap {
		swap = false
		for i := 0; i < unsortedUntilIndex; i++ {
			first  := numbers[i]
			second := numbers[i + 1]

			if first > second { numbers[i] = second; numbers[i + 1] = first; swap = true }
		}
		unsortedUntilIndex--
	}

	return numbers
}

File: seeker/snippet/BubbleSort.java
//date: 2022-10-14T17:12:30Z
//url: https://api.github.com/gists/f2387fa7c70c86dfecfc738ae5d7c710
//owner: https://api.github.com/users/LastKnell

package Arrays.Sorting;

public class BubbleSort {
    /*
     * This method takes a int array as input
     * Then sorts the array using BUBBLE SORT algorithm
     * Then returns the sorted array
     */
    public static int[] sort(int[] arr) {
        int n = arr.length;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
        return arr;
    }
}